src/test/java/jimmy/mcgymmy/logic/commands/AddCommandTest.java

```diff
@@ -42,6 +45,25 @@ public void execute_foodAcceptedByModel_addSuccessful() throws Exception {
         assertEquals(Arrays.asList(validFood), modelStub.foodAdded);
     }
 
+    @Test
+    public void execute_validFoodWithTag_addSuccessful() throws Exception {
+        ModelStubAcceptingFoodAdded modelStub = new ModelStubAcceptingFoodAdded();
+        Food validFood = new FoodBuilder().withCarb("12345").withTags("hello").build();
+        AddCommand command = new AddCommand();
+        command.setParameters(
+            new CommandParserTestUtil.ParameterStub<>("n", validFood.getName()),
+            new CommandParserTestUtil.OptionalParameterStub<>("p", validFood.getProtein()),
+            new CommandParserTestUtil.OptionalParameterStub<>("f", validFood.getFat()),
+            new CommandParserTestUtil.OptionalParameterStub<>("c", validFood.getCarbs()),
+            new CommandParserTestUtil.OptionalParameterStub<>("t", new Tag("hello"))
+        );
+
+        CommandResult commandResult = command.execute(modelStub);
+
+        assertEquals(String.format(AddCommand.MESSAGE_SUCCESS, validFood), commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validFood), modelStub.foodAdded);
```

Comment:
Might be able to use CommandTestUtil here to reduce code duplication.
Can compare the tags after the CommandTestUtil.assertCommandSuccess

Reply:
Fixed